### ISSUE TITLE
UX proposal - Full Screen Modal

### ISSUE THREAD

**Issue by rossbulat** [polkadot-js/apps#1201]
`[rossbulat: polkadot JS milestone 1 proposals]`

#### Goal
UX Elements to fill voids in Polkadot form management and general design across packages.

#### Nature of Work
Iterative form development 

#### Specifications
- [x] All elements are designed to be responsive for all screen sizes
- [x] All elements are structural components; all can be `ui-app` components to be imported elsewhere in apps
- [x] `styled-components` styling comes first, least reliance on external styles to prevent bloat
- [x] Designed to be fully typed with generic typing where appropriate

#### Discussion Goals
Positive consensus for each element, additions or amendments where agreed upon. Where one does not support an element, offer an alternative solution.

## Full Screen Modal `FullModal`
A bare-bones element to overlay content on a full screen basis, with access to the glossary.

<img width="868" alt="FullScreenModal" src="https://user-images.githubusercontent.com/13929023/58313411-b5ce2d80-7e37-11e9-9104-6dd4129c04f7.png">

#### Justification
Current modal is ugly, does not reach design standards and is overly complicated in the amount of classes being attached to it.

#### Styling Considerations
- Flex box-based layout where header and footer are fixed, (footer optional for call-to-action).
- Ability to call the glossary, as tooltips are also present inside modals.
- Close button situated on left side to not interfere with glossary and main content

**Comment by jacogr** [comment]
This looks like a solution looking for a problem. With all the modals, we have not run into one where it was "o, this really should be full-screen". (I certainly have ideas, or rather long-fetched "would it be cool if" uses in the long term, but as it stands cannot really see spending effort on it and having more code to maintain as a result.)

Additionally, need to be careful when doing this full-screen -

- does it really need to be a modal (or rather just a full-screen page with fwd/back navs)
- typically when doing this, there would be a resulting transaction, which is modal-on-modal - not the best UX.

Basically, without an existing reason on "functionality xyz can be better with this", it is not an appetizing solution for any known problem. (Apart from a broad "the current modals are ugly" - don't disagree, there are certainly a lot of things that can and should be done on a per-modal basis to make it easy, simple and straight forward.)

TL:DR - So looking at closing this without a specific use-case.

**Comment by jacogr** [comment]
Have just been through all the modals in the last 2 days again - the largest one we have is the signer (which also varies in size, from one-liners to complex structures), however there we also don't have need for full-screen atm. 

(We certainly added a bunch over the last while, staking, transfer, address/account create, etc. - most actions apart from the more-technical areas are modal-driven now)

I think not having a need _may_ change for an hand-holding first-run intro if we ever get there, however the fullscreen option from https://react.semantic-ui.com/modules/modal/ is config away. (We are consistently applying the "large" size for all modals atm, so no surprises along the way in any area)
